modules/bq-log-alerting/main.tf

```diff
@@ -0,0 +1,158 @@
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "enable_securitycenter_api" {
+  project            = var.logging_project
+  service            = "securitycenter.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "enable_pubsub_api" {
+  project            = var.logging_project
+  service            = "pubsub.googleapis.com"
+  disable_on_destroy = false
+}
+
+#--------------------------#
+# Service account creation #
+#--------------------------#
+resource "random_string" "service_account" {
+  length  = 12
+  upper   = "false"
+  number  = "false"
+  special = "false"
+}
+
+resource "google_service_account" "gcf_service_account" {
+  project      = var.logging_project
+  account_id   = "cloudfunction-${random_string.service_account.result}"
+  display_name = "Cloud Function Service Account"
+
+  depends_on = [
+    google_project_service.enable_cloudscheduler_api,
+    google_project_service.enable_cloudbuild_api,
+    google_project_service.enable_cloudfunctions_api,
+  ]
+}
+
+#--------------------------------#
+# Service account IAM membership #
+#--------------------------------#
+resource "google_project_iam_member" "gcf-big-query" {
+  project = var.logging_project
+  role    = "roles/bigquery.admin"
+  member  = "serviceAccount:${google_service_account.gcf_service_account.email}"
+}
+
+resource "google_organization_iam_member" "gcf-security-findings" {
+  org_id = var.org_id
+  role   = "roles/securitycenter.findingsEditor"
+  member = "serviceAccount:${google_service_account.gcf_service_account.email}"
+}
+
+#-----------------------------------------#
+# Security Command Center Source creation #
+#-----------------------------------------#
+resource "google_scc_source" "bq_log_alerts" {
+  count        = var.source_name == "" ? 1 : 0
+  display_name = "BQ Log Alerts"
+  organization = var.org_id
+  description  = "Findings from BQ Alerting Solution"
+
+  depends_on = [
+    google_project_service.enable_securitycenter_api
+  ]
+}
+
+#------------------------#
+# Bigquery views dataset #
+#------------------------#
+resource "google_bigquery_dataset" "views_dataset" {
+  dataset_id    = "views"
+  friendly_name = "Log Views"
+  description   = "Log view dataset"
+  location      = "US"
+  project       = var.logging_project
+
+  labels = {
+    env = "default"
+  }
+
+  depends_on = [
+    google_project_service.enable_bigquery_api
+  ]
+}
+
+#-----------------------------#
+# Scheduled function creation #
+#-----------------------------#
+module "bq-log-alerting" {
+  source                         = "terraform-google-modules/scheduled-function/google"
+  version                        = "1.5.0"
+  project_id                     = var.logging_project
+  job_name                       = "bq-alerts-event-trigger"
+  job_description                = "publish to pubsub to trigger cloud function"
+  job_schedule                   = "*/15 * * * *"
```

Comment:
Should this be configurable?

Reply:
We can add the pair `job_schedule` and `message_data` to the variables with the current values as default. but we to explain which are the impacts of changing them and that the `message_data` needs to have a value that is larger then the `job_schedule` interval so that findings are not lost in the ingestion process.

Should we add them as variables and add a section in the README for the effect of changing them? @bharathkkb

Reply:
ack, let chat during tomorrow's standup